File: _프로그래머스/01-20/8. 하샤드 수.py
"""
* Title: 하샤드 수
https://school.programmers.co.kr/learn/courses/30/lessons/12947
"""


def solution(x):
    arr = list(str(x))
    sum = 0
    answer = True

    for i in range(len(arr)):
        sum += int(arr[i])

    if x % sum == 0:
        answer = True
    else:
        answer = False

    return answer

File: _프로그래머스 js/01-20/8. 하샤드 수.js
/*
* Title: 하샤드 수
https://school.programmers.co.kr/learn/courses/30/lessons/12947
*/

const solution = (x) => {
    const arr = Array.from(String(x), Number);
    const sum = arr.reduce((a, b) => a + b, 0);
    let answer = true;

    if (x % sum !== 0) {
        answer = false;
    }

    return answer;
}

// true
console.log(solution(10))

// false
console.log(solution(13))